src/v/storage/disk_log_impl.cc

```diff
@@ -572,15 +572,32 @@ offset_stats disk_log_impl::offsets() const {
     // with reverse iterators, so we manually find the iterator
     segment_set::type end;
     segment_set::type term_start;
+    segment_set::type last_committed;
     for (int i = (int)_segs.size() - 1; i >= 0; --i) {
         auto& seg = _segs[i];
         if (!seg->empty()) {
             if (!end) {
                 end = seg;
             }
+
+            /**
+             * move back until we find a segment with initialized committed
+             * offset
+             */
+            if (
+              !last_committed
+              || last_committed->offsets().committed_offset
+                   < model::offset{0}) {
+                last_committed = seg;
+            }
+            if (
+              last_committed->offsets().committed_offset
```

Comment:
This looks like it will select a segment with negative (ie uninitialized) offset. Is that intentional?